### ISSUE TITLE
exim4: input sent without waiting for greeting

### ISSUE THREAD

**Issue by lerouxb** [Marak/node_mailer#19]
This is with the latest latest code from git, in /var/log/exim4/rejectlog:

2011-01-28 10:48:33 SMTP protocol synchronization error (input sent without waiting for greeting): rejected connection from H=localhost [127.0.0.1] input="helo localhost\r\nmail from: xxx@yyy.com\r\n"

**Comment by Marak** [comment]
Are you 100% sure you are using the latest version? There is a specific fix to wait for HELO before starting to send the message.

**Comment by lerouxb** [comment]
I pulled the code within the last 24 hours specifically because i saw that change and thought it might help.

**Comment by Marak** [comment]
Did you do a git pull or an npm install? If you just pulled the code that won't update the NPM package unless you npm link the folder.

**Comment by lerouxb** [comment]
I'm actually not using npm in this context - I'm doing a require() directly on the file. (I copied it to a "vendor" directory.) My first assumption was that it was still the old file and i already checked, double-checked, etc. I am looking at my source tree and the updated file is there :)

I ended up giving up and now I'm just using sendmail.

I reported it hoping that someone else would/could check in exim too. Perhaps I am just losing my mind because the code looks like it should work...
